smash/core/simulation/optimize/optimize.py

```diff
@@ -984,3 +794,345 @@ def _bayesian_optimize(
+    parameters: ParametersDT,
+    wrap_options: OptionsDT,
+    wrap_returns: ReturnsDT,
+    return_options: dict,
+) -> dict:
+    wrap_parameters_to_control(
+        model.setup,
+        model.mesh,
+        model._input_data,
+        parameters,
+        wrap_options,
+    )
+
+    x0 = parameters.control.x.copy()
+
+    l_control = np.where(parameters.control.l == -99, None, parameters.control.l)
+    u_control = np.where(parameters.control.u == -99, None, parameters.control.u)
+
+    cb = _OptimizeCallback(wrap_options.comm.verbose)
+
+    res_optimize = scipy_minimize(
+        _optimize_problem,
+        x0,
+        args=(model, parameters, wrap_options, wrap_returns, cb),
+        method="l-bfgs-b",
+        jac=_jac_optimize_problem,
+        bounds=tuple(zip(l_control, u_control)),
+        callback=cb.callback,
+        options={
+            "maxiter": wrap_options.optimize.maxiter,
+            "ftol": 2.22e-16 * wrap_options.optimize.factr,
+            "gtol": wrap_options.optimize.pgtol,
+            "disp": False,  # TODO: change this with logger for multiple display levels
+        },
+    )
+
+    cb.termination(res_optimize)
+
+    ret = {}
+
+    if "control_vector" in return_options["keys"]:
+        ret["control_vector"] = parameters.control.x.copy()
+
+    if "iter_cost" in return_options["keys"]:
+        ret["iter_cost"] = cb.iter_cost
+
+    if "iter_projg" in return_options["keys"]:
+        ret["iter_projg"] = cb.iter_projg
+
+    if "serr_mu" in return_options["keys"]:
+        ret["serr_mu"] = model.get_serr_mu().copy()
+
+    if "serr_sigma" in return_options["keys"]:
+        ret["serr_sigma"] = model.get_serr_sigma().copy()
+
+    return ret
+
+
+def _sbs_optimize(
+    model: Model,
+    parameters: ParametersDT,
+    wrap_options: OptionsDT,
+    wrap_returns: ReturnsDT,
+    return_options: dict,
+) -> dict:
+    wrap_parameters_to_control(
+        model.setup,
+        model.mesh,
+        model._input_data,
+        parameters,
+        wrap_options,
+    )
+
+    n = parameters.control.n
+
+    sdx = np.zeros(n)
+
+    y_wa = parameters.control.x.copy()
+    z_wa = np.copy(y_wa)
+
+    l_wa = parameters.control.l.copy()
+    u_wa = parameters.control.u.copy()
+
+    wrap_forward_run(
+        model.setup,
+        model.mesh,
+        model._input_data,
+        parameters,
+        model._output,
+        wrap_options,
+        wrap_returns,
+    )
+
+    gx = model._output.cost
+    ga = gx
+    clg = 0.7 ** (1 / n)
+    ddx = 0.64
+    dxn = ddx
+    ia = iaa = iam = -1
+    jfaa = 0
+    nfg = 1
+
+    ret = {}
+
+    message = "STOP: TOTAL NO. OF ITERATION EXCEEDS LIMIT"
+
+    if wrap_options.comm.verbose:
+        print(
+            f"    At iterate{str(0).rjust(7)}    nfg = {str(nfg).rjust(4)}"
+            f"    J = {gx:14.6f}    ddx = {ddx:5.2f}"
+        )
+
+    if "iter_cost" in return_options["keys"]:
+        ret["iter_cost"] = np.zeros(wrap_options.optimize.maxiter + 1)
```

Comment:
Si la simulation stop avant le maximum d'itération, le tableau aura des zeros en trop